keanu-project/src/main/java/io/improbable/keanu/vertices/dbltensor/TensorShapeValidation.java

```diff
@@ -0,0 +1,108 @@
+package io.improbable.keanu.vertices.dbltensor;
+
+import java.util.Arrays;
+import java.util.Set;
+
+import static java.util.stream.Collectors.toSet;
+
+public class TensorShapeValidation {
+
+    private TensorShapeValidation() {
+    }
+
+    /**
+     * This is a common function to check that tensors are either
+     * the same shape of the proposal in question OR scalar.
+     *
+     * @param proposalShape the tensor shape being validated
+     * @param tensors       the tensors being validated against
+     * @throws IllegalArgumentException if there is more than one non-scalar shape OR if the non-scalar shape does
+     *                                  not match the proposal shape.
+     */
+    public static void checkTensorsMatchNonScalarShapeOrAreScalar(int[] proposalShape, Tensor... tensors) {
```

Comment:
I think you're fine to call this `checkTensorsMatchOrAreScalar`, it will read a bit nicer too